src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -96,46 +96,49 @@ export class TransactionApi implements ITransactionApi {
     await this.cacheService.deleteByKey(cacheKey);
   }
 
-  async getDataDecoded(data: string, to?: string): Promise<DataDecoded> {
+  async getDataDecoded(args: {
+    data: string;
+    to?: string;
+  }): Promise<DataDecoded> {
     try {
       const url = `${this.baseUrl}/api/v1/data-decoder/`;
       const { data: dataDecoded } = await this.networkService.post(url, {
-        data,
-        to,
+        data: args.data,
+        to: args.to,
```

Comment:
You can use `data` directly here instead of create a new object.

Reply:
> You can use `data` directly here instead of create a new object.

You meant `args` here right?

Reply:
On a second thought I have some concerns regarding using `args` directly. What if there is a new arg that is used for other purposes (e.g. a cache key)? If we use `args` directly we would send data to the Safe Transaction Service that might not be required (and might provide data that is internal to the CGW instance). Wdyt?

Reply:
> You meant args here right?

Sorry, typo.

> What if there is a new arg that is used for other purposes (e.g. a cache key)? If we use `args` directly we would send data to the Safe Transaction Service that might not be required (and might provide data that is internal to the CGW instance). Wdyt?

If we were to later add new arguments, it would then be tested. However, it is also more explicit as is. I have no strong opinion.

Reply:
> > You meant args here right?
> 
> Sorry, typo.
> 
> > What if there is a new arg that is used for other purposes (e.g. a cache key)? If we use `args` directly we would send data to the Safe Transaction Service that might not be required (and might provide data that is internal to the CGW instance). Wdyt?
> 
> If we were to later add new arguments, it would then be tested. However, it is also more explicit as is. I have no strong opinion.

I'd say that we can use the spread operator or the args bundle throughout the internal functions of the CGW but when interacting with external services, to be more explicit on what we are sending (to avoid potential leaks).

Reply:
> I'd say that we can use the spread operator or the args bundle throughout the internal functions of the CGW but when interacting with external services, to be more explicit on what we are sending (to avoid potential leaks).

Yeah, I totally support this 💯